pkg/service/types.go

```diff
@@ -22,12 +22,19 @@ const (
 	RemoteCluster
 )
 
+// ProviderMapper describes any entity that can be mapped to a service.Provider based on NamespacedKey
+type ProviderMapper interface {
```

Comment:
I fear this may be a little more clever than it needs to be. Can we get rid of this interface? We can still keep the NamespacedKey(), but I don't see a need for an interface, especially since we have no plans to have anything else implement it

Reply:
The only reason I created it was to start a conversation on possibly reducing the coupling to MeshService throughout the code. But I see your point: if we're fine with MeshService continuing to be the contract between providers and mesh configuration, then I'm good with taking a YAGNI approach and removing it